# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/ipc_message.h
# Commit: 451508713a62
# Full Hash: 451508713a625beaaa9fcbbf06c65ca312dc34fb
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 878722e87e62 -r 451508713a62 ipc/chromium/src/chrome/common/ipc_message.h
--- a/ipc/chromium/src/chrome/common/ipc_message.h	Tue Jun 22 18:17:23 2021 +0000
+++ b/ipc/chromium/src/chrome/common/ipc_message.h	Tue Jun 22 18:17:23 2021 +0000
@@ -282,7 +282,8 @@
   size_t GetSizeIfSerialized() const override { return size(); }
 
   void WriteFooter(const void* data, uint32_t data_len);
-  [[nodiscard]] bool ReadFooter(void* buffer, uint32_t buffer_len);
+  [[nodiscard]] bool ReadFooter(void* buffer, uint32_t buffer_len,
+                                bool truncate);
   uint32_t FooterSize() const;
 
   // Used for async messages with no parameters.